main/lsp/LSPTypechecker.cc

```diff
@@ -128,6 +132,9 @@ bool LSPTypechecker::typecheck(LSPFileUpdates updates, WorkerPool &workers) {
     } else {
         committed = runSlowPath(move(updates), workers, /* cancelable */ true);
     }
+
+    errorReporter.endEpoch(updates.epoch, committed);
```

Comment:
Ditto.

Reply:
Would it be cleaner to use [RAII](https://www.tomdalling.com/blog/software-design/resource-acquisition-is-initialisation-raii-explained/) for epochs? e.g.

```
{
  ErrorEpoch epoch(errorReporter, updates.epoch); // calls start epoch
  epoch.committed = runSlowPath(...);
  // destructor calls end epoch with `committed`
}
```

We can define `ErrorEpoch` in the `ErrorReporter` header.